src/Semantics.js

```diff
@@ -260,39 +360,7 @@ Semantics.prototype.addOperationOrAttribute = function(type, nameAndFormalArgs,
   // Create the action dictionary for this operation / attribute that contains a `_default` action
   // which defines the default behavior of iteration, terminal, and non-terminal nodes...
   var realActionDict = {
-    _default: function(children) {
-      var self = this;
-      var thisThing = this._semantics[typePlural][name];
-      var args = thisThing.formals.map(function(formal) {
-        return self.args[formal];
-      });
-
-      if (this.isIteration()) {
-        // This CST node corresponds to an iteration expression in the grammar (*, +, or ?). The
-        // default behavior is to map this operation or attribute over all of its child nodes.
-        return children.map(function(child) { return doIt.apply(child, args); });
-      }
-
-      if (this.isTerminal()) {
-        // This CST node corresponds to a terminal expression in the grammar (e.g., "+"). The
-        // default behavior is to return that terminal's primitive value.
-        return this.primitiveValue;
-      }
-
-      // This CST node corresponds to a non-terminal in the grammar (e.g., AddExpr). The fact that
-      // we got here means that this action dictionary doesn't have an action for this particular
-      // non-terminal or a generic `_nonterminal` action.
-      if (children.length === 1) {
-        // As a convenience, if this node only has one child, we just return the result of
-        // applying this operation / attribute to the child node.
-        return doIt.apply(children[0], args);
-      } else {
-        // Otherwise, we throw an exception to let the programmer know that we don't know what
-        // to do with this node.
-        throw new Error(
-            'Missing semantic action for ' + this.ctorName + ' in ' + name + ' ' + type);
-      }
-    }
+    _default: newDefaultAction(typePlural, type, name, doIt)
```

Comment:
Instead of the string comparison, what if you just save the result of `newDefaultAction`, and either pass that in to the Operation/Attribute constructor, or just set it as a property later?

E.g.:
```
var builtInDefault = newDefaultAction(...);
var realActionDict = {_default: builtInDefault};
...
this[typePlural][name] = type === 'operation' ?
       new Operation(name, inheritedFormals, newActionDict, builtInDefault) :
       new Attribute(name, newActionDict, builtInDefault);
```
Then you can use simple equality to ensure that you don't include that in the recipe.